nidirect_workflow/nidirect_workflow.module

```diff
@@ -11,3 +20,59 @@ function nidirect_workflow_page_attachments(array &$attachments) {
   // Attach extra custom css for admin menu.
   $attachments['#attached']['library'][] = 'nidirect_workflow/admin.css';
 }
+
+/**
+ * Implements hook_entity_presave().
+ */
+function nidirect_workflow_entity_presave(EntityInterface $entity) {
```

Comment:
A suggestion: consider the use of an early return pattern to limit the amount of depth/indentation in the code here.

So instead of:

```
function thingy() {
  if ($thing == 'a-certain-value') {
    // more checks/things to do...
    return $value;
  }
}
```

try:

```
function thingy() {
  if ($thing != 'a-specific-value') {
    return;
  }

  // Anything from here is useful code/contextually relevant. Otherwise, we bail early. Bonus points for not needing an extra level of indentation, keeping it more readable.

  // Return a thing.
  return $thing;
}
```

Reply:
Hmm, yes, interesting (and debatable).

I think that I'm still stuck in the old school idea of "only one return in a function" but some Googling suggests that this is a bit outdated now. I must say though that I am still reluctant to have many return statements in a long and complex function (perhaps there shouldn't be any long and complex functions anywhere, that's a separate discussion?!).

I do agree that this code looks a bit ugly so will go with your suggestion here.

Reply:
Have had another look at this and actually think I'll leave it as is, early returns still don't sit well with me !